### PR TITLE
Trigger back pressure signals and in-flight command controls

### PR DESCRIPTION
**Pull request by carlpulley-da**
- [x] crash when number of in-flight commands is excessive
- [x] inject configuration into Daml `initialState` function
- [x] refactor batch trigger `initialState` to take a record of arguments (to simplify any future argument extensions)
- [x] Daml tests for new behaviours
- [x] address backwards compatibility issues with config injection into Daml code

### CODE REVIEW

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala [hunk 1]
```diff
@@ -105,10 +105,11 @@ private[lf] final case class Trigger(
   def initialStateArguments(
       parties: TriggerParties,
       acs: Seq[CreatedEvent],
+      triggerConfig: TriggerRunnerConfig,
       converter: Converter,
   ): Array[SValue] = {
     if (defn.version >= Trigger.Version.`2.5.1`) {
-      Array(converter.fromTriggerSetupArguments(parties, acs).orConverterException)
+      Array(converter.fromTriggerSetupArguments(parties, acs, triggerConfig).orConverterException)
```
Comment (carlpulley-da) [inline triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala:112]: TODO: what about backwards compatibility?

File: triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/FailureTesting.scala [hunk 2]
```diff
@@ -30,14 +35,18 @@ final class FailureTesting
...
     participants = singleParticipant(
       ApiServerConfig.copy(
         timeProviderType = TimeProviderType.Static
       )
     )
   )
 
+  override protected def triggerRunnerConfiguration: TriggerRunnerConfig =
+    super.triggerRunnerConfiguration
+      .copy(inFlightCommandBackPressureCount = 20, inFlightCommandOverflowCount = 200)
```
Comment (carlpulley-da) [inline triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/FailureTesting.scala:48]: Arbitrary low values to ensure trigger overflow limits are hit in ITs

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunnerConfig.scala [hunk 3]
```diff
@@ -34,6 +40,8 @@ object TriggerRunnerConfig {
       maxRetries = 6,
       maxSubmissionRequests = 100,
       maxSubmissionDuration = 5.seconds,
+      inFlightCommandBackPressureCount = 1000,
+      inFlightCommandOverflowCount = 10000,
```
Comment (carlpulley-da) [inline triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunnerConfig.scala:44]: From a backwards compatibility PoV: should we have a CLI flag to disable the use of overflows?

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala [hunk 4]
```diff
@@ -1007,22 +1010,50 @@ private[lf] class Runner private (
...
-        retrying(
-          initialTries = triggerConfig.maxRetries,
-          backoff = overloadedRetryDelay,
-          parallelism = triggerConfig.parallelism,
-          retryableSubmit,
-          submit,
+    // Used to defend (by raising an unhandled exception) against trigger rules that emit excessive numbers of command submission requests
+    val overflowFlow: TriggerContextualFlow[SubmitRequest, SubmitRequest, NotUsed] =
+      TriggerContextualFlow[SubmitRequest]
+        .map {
```
Comment (carlpulley-da) [inline triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala:1016]: Need to use a map here to ensure that the exception/failure is propagated

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunnerConfig.scala (hunk 3 above)
Comment (remyhaemmerle-da) [reply triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunnerConfig.scala:44]: I think it is a good idea to have a flag, not necessary for backward compatibility.

**Review by remyhaemmerle-da — APPROVED** [review]
Nice.

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunnerConfig.scala (hunk 3 above)
Comment (carlpulley-da) [reply triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunnerConfig.scala:46]: I've added a flag to disable overflow checking and, in addition, I've added the ability to alter the overflow count from the CLI.

**Pull request merged**